### DISCUSSION TITLE
0.8.0

### DISCUSSION THREAD

**Discussion by mitsuhiko** [astral-sh/rye#333] [Announcements]
## What's Changed

- Rye for now prefers `>=` over `~=` for newly added dependencies.

- The workspace member declaration is now platform independent.  If `members` is
  now explicitly set to an empty list it will not fall back to auto discovery.  #331

- `rye add` now pins versions with `==` instead of `~=` when the version of the
  package does not use at least two components.  This means that for instance it
  will now correctly use `openai-whisper==20230314` rather than
  `openai-whisper~=20230314` which is not actually satisfiable.  #328

- `rye install` now lets you install dependencies into the tool's virtualenv
  during installation that are undeclared via the new `--extra-requirement`
  option.  #326

- Improved handling of relative path installations by setting `PROJECT_ROOT`
  the same way as PDM does.  #321

- Workspaces will now never discover `pyproject.toml` files in any dot
  directories. (Name starting with `.`)  #329

- Fixed `rye build` not working correctly on Windows.  #327

**Full Changelog**: https://github.com/mitsuhiko/rye/compare/0.7.0...0.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mitsuhiko/rye/releases/tag/0.8.0'>0.8.0</a>.</em>

**Comment by bluss** [comment]
Would it be possible to give nicknames to ">=" and "~=" so that we could use them easily from `rye add`?  Maybe one of these syntaxes could work, user would use it to ask for a specific type of version restriction:

- `rye add --pin=tilde numpy`
- `rye add "numpy ~="`

**Reply by mitsuhiko** [reply]
Filed under #340.

**Reply by bluss** [reply]
Nice!
